File: back-end/go/internal/models/promotion.go
package models

import (
    "go.mongodb.org/mongo-driver/bson/primitive"
)

type Promotion struct {
    ID             primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"`
    Code           string             `json:"code"`
    Title          string             `json:"title"`
    Description    string             `json:"description"`
    MaxDiscountPct float32            `json:"max_discount_pct"`
    Products       []string           `json:"products,omitempty"`
}

File: back-end/nodejs/internal/models/promotion.js
var mongoose = require('mongoose');

var schema = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    code: { 
        type: String, 
        required: true 
    },
    title: {
        type: String, 
        required: true 
    },
    description: {
        type: String, 
        required: true 
    },
    max_discount_pct: {
        type: Number, 
        required: true 
    },
    products: {
        type: [{
            type: String
        }],
        require: false
    }
});

var promotion = new mongoose.model('Promotion', schema);

module.exports = promotion;